pkg/csi_driver/multishare_controller.go

```diff
@@ -440,14 +470,19 @@ func (m *MultishareController) generateNewMultishareInstance(instanceName string
 		KmsKeyName:  kmsKeyName,
 		Labels:      labels,
 		Description: generateInstanceDescFromEcfsDesc(m.ecfsDescription),
-	}, nil
+	}
+	if m.featureMaxSharePerInstance {
+		f.MaxShareCount = maxShareCount
+	}
+	return f, nil
 }
 
 func generateNewShare(name string, parent *file.MultishareInstance, req *csi.CreateVolumeRequest) (*file.Share, error) {
 	if parent == nil {
 		return nil, status.Error(codes.Internal, "parent mulishare instance is empty")
 	}
-	targetSizeBytes, err := getShareRequestCapacity(req.CapacityRange)
+	// The share size request is already validated in CreateVolume call
+	targetSizeBytes, err := getShareRequestCapacity(req.CapacityRange, util.MinShareSizeBytes, util.MaxShareSizeBytes)
```

Comment:
don't we need an if clause here to pass in `util.MinShareSizeConfigurableBytes` and user-specified max range here?

Reply:
yes it will be in the next PR

Reply:
as mentioned in the comment, by the time we reach to the `generateNewShare` function we have already validated the range. So we do not need a strict range change. Hence I have kept the function as it is.

Reply:
what I mean is that if the feature is turned on, and user ask for 10Gi, here the `util.MinShareSizeBytes` passed into `getShareRequestCapacity` would mean the 10Gi is below the min size and produced error right?